pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -109,6 +110,24 @@ type EventSourceSpec struct {
 	Pulsar map[string]PulsarEventSource `json:"pulsar,omitempty" protobuf:"bytes,27,opt,name=pulsar"`
 	// Generic event source
 	Generic map[string]GenericEventSource `json:"generic,omitempty" protobuf:"bytes,28,rep,name=generic"`
+	// Replicas is the event source deployment replicas
+	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,29,opt,name=replicas"`
+}
+
+func (e EventSourceSpec) GetReplicas() int32 {
+	if e.Replicas == nil && e.DeprecatedReplica == nil {
```

Comment:
This is just complicated enough to warrant a unit test.

Reply:
Test cases added.